maven-plugin/src/main/java/org/openmrs/maven/plugins/BuildDistro.java

```diff
@@ -587,4 +592,85 @@ public boolean accept(File dir, String name) {
+	 * 
+	 * @param userDir
+	 * @param distroProperties
+	 * @throws MojoExecutionException
+	 */
+	private void parseContentProperties(File userDir, DistroProperties distroProperties) throws MojoExecutionException {
+		File contentFile = new File(userDir, CONTENT_PROPERTIES);
+		if (!contentFile.exists()) {
+			log.info("No content.properties file found in the user directory.");
+			return;
+		}
+
+		Properties contentProperties = new Properties();
+		try (BufferedReader reader = new BufferedReader(new FileReader(contentFile))) {
+			contentProperties.load(reader);
+		} catch (IOException e) {
+			throw new MojoExecutionException("Failed to read content.properties file", e);
+		}
+
+		for (String dependency : contentProperties.stringPropertyNames()) {
+			String versionRange = contentProperties.getProperty(dependency);
+			String distroVersion = distroProperties.get(dependency);
+
+			if (distroVersion == null) {
+				String latestVersion = findLatestMatchingVersion(dependency, versionRange);
+				if (latestVersion == null) {
+					throw new MojoExecutionException("No matching version found for dependency " + dependency);
+				}
+
+				distroProperties.add(dependency, latestVersion);
+			} else {
+				checkVersionInRange(dependency, versionRange, distroVersion);
+			}
+		}
+	}
+
+	private String findLatestMatchingVersion(String dependency, String versionRange) throws MojoExecutionException {
+		return distroHelper.findLatestMatchingVersion(dependency, versionRange);
+	}
+
+	/**
+     * Checks if the version from distro.properties satisfies the range specified in content.properties.
+     * Throws an exception if there is a mismatch.
+     *
+     * @param contentDependencyKey       the dependency key
+     * @param contentDependencyVersionRange the version range from content.properties
+     * @param distroPropertyVersion      the version from distro.properties
+     * @throws MojoExecutionException if there is a version mismatch
+     */
+	private void checkVersionInRange(String contentDependencyKey, String contentDependencyVersionRange, String distroPropertyVersion) throws MojoExecutionException {
+		Semver semverVersion = new Semver(distroPropertyVersion, Semver.SemverType.NPM);
+		String[] ranges = contentDependencyVersionRange.split(",");
+		boolean inRange = false;
+
+		for (String range : ranges) {
+			range = range.trim();
+			try {
+				if (range.startsWith("^") || range.startsWith("~")) {
+					inRange = semverVersion.satisfies(range);
+				} else {
+					Semver rangeVersion = new Semver(range, Semver.SemverType.NPM);
+					inRange = semverVersion.diff(rangeVersion) == Semver.VersionDiff.NONE;
+				}
+				if (inRange) break;
+			} catch (SemverException e) {
+				throw new MojoExecutionException("Invalid version range format for " + contentDependencyKey + ": " + contentDependencyVersionRange, e);
+			}
+		}
+
+		if (!inRange) {
+			throw new MojoExecutionException(
```

Comment:
Error message should probably also include a way of identifying the content package in question since a distro may have several.

Reply:
Oh, makes perfect sense